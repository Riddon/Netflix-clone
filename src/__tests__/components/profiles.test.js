import React from 'react';
import { render } from '@testing-library/react';
import { Profiles } from "../../components/";

describe('<Profiles />', () => {
    it('renders the <Profiles/> with populated data',  () => {
        const { container, getByText, getByTestId } = render(
            <Profiles>
                <Profiles.Title>
                    Who`s watching?
                </Profiles.Title>
                <Profiles.List>
                    <Profiles.User onClick={() => {}}>
                        <Profiles.Picture src="/images/karlitto.png" data-testid={'getByTestId'}/>
                        <Profiles.Name>Karlitto</Profiles.Name>
                    </Profiles.User>
                </Profiles.List>
            </Profiles>
        );

        expect(getByText('Who`s watching?')).toBeTruthy();
        expect(getByText('Karlitto')).toBeTruthy();
        expect(getByTestId('getByTestId')).toBeTruthy();

        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders the <Profiles/> without src',  () => {
        const { container, getByText, getByTestId } = render(
            <Profiles>
                <Profiles.Title>
                    Who`s watching?
                </Profiles.Title>
                <Profiles.List>
                    <Profiles.User onClick={() => {}}>
                        <Profiles.Picture src={false} data-testid={'getByTestId'}/>
                        <Profiles.Name>Karlitto</Profiles.Name>
                    </Profiles.User>
                </Profiles.List>
            </Profiles>
        );

        expect(getByText('Who`s watching?')).toBeTruthy();
        expect(getByText('Karlitto')).toBeTruthy();
        expect(getByTestId('getByTestId').src).toContain('/images/misc/loading.gif');

        expect(container.firstChild).toMatchSnapshot();
    });
});