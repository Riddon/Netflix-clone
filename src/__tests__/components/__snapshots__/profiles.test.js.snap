// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profiles /> renders the <Profiles/> with populated data 1`] = `
<div
  class="profiles__Container-bxtkkv-0 lfIDZX"
>
  <h1
    class="profiles__Title-bxtkkv-1 kjBRZT"
  >
    Who\`s watching?
  </h1>
  <ul
    class="profiles__List-bxtkkv-2 livHdo"
  >
    <li
      class="profiles__Item-bxtkkv-5 bsnzsB"
    >
      <img
        class="profiles__Picture-bxtkkv-3 hMyKjx"
        data-testid="getByTestId"
        src="/images/users//images/karlitto.png.png"
      />
      <p
        class="profiles__Name-bxtkkv-4 fQlGYa"
      >
        Karlitto
      </p>
    </li>
  </ul>
</div>
`;

exports[`<Profiles /> renders the <Profiles/> without src 1`] = `
<div
  class="profiles__Container-bxtkkv-0 lfIDZX"
>
  <h1
    class="profiles__Title-bxtkkv-1 kjBRZT"
  >
    Who\`s watching?
  </h1>
  <ul
    class="profiles__List-bxtkkv-2 livHdo"
  >
    <li
      class="profiles__Item-bxtkkv-5 bsnzsB"
    >
      <img
        class="profiles__Picture-bxtkkv-3 hMyKjx"
        data-testid="getByTestId"
        src="/images/misc/loading.gif"
      />
      <p
        class="profiles__Name-bxtkkv-4 fQlGYa"
      >
        Karlitto
      </p>
    </li>
  </ul>
</div>
`;
